src/Shouldly/Internals/ShouldThrowAssertionContext.cs

```diff
@@ -4,18 +4,18 @@ namespace Shouldly
 {
     internal class ShouldThrowAssertionContext : ShouldlyAssertionContext
     {
-        public string ExceptionMessage { get; private set; }
+        public string? ExceptionMessage { get; private set; }
 
         public bool IsAsync { get; private set; }
 
 #if StackTrace
         /// <summary>
         /// Only pass stacktrace if asynchronous
         /// </summary>
-        internal ShouldThrowAssertionContext(object expected, object actual = null, string exceptionMessage = null,
+        internal ShouldThrowAssertionContext(object? expected, object? actual = null, string? exceptionMessage = null,
             bool isAsync = false,
-            System.Diagnostics.StackTrace stackTrace = null,
-            [CallerMemberName] string shouldlyMethod = null) : base(shouldlyMethod, expected, actual, stackTrace)
+            System.Diagnostics.StackTrace? stackTrace = null,
+            [CallerMemberName] string shouldlyMethod = null!) : base(shouldlyMethod, expected, actual, stackTrace)
```

Comment:
I don't think there's a situation where the default value `null` could be used.
https://docs.microsoft.com/en-us/dotnet/api/system.runtime.compilerservices.callermembernameattribute#remarks